app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,68 @@
+class MarketsController < ApplicationController
+  def index
+    @markets = Market.all
+  end
+
+  def login
+    id = params.permit(:login_id)[:login_id].to_i
+
+    if (id <= Market.last.id && id > 0)
+      redirect_to "/markets/#{ id }/dashboard"
+    else
+      redirect_to "/markets/market_not_found"
```

Comment:
I noticed you doing this in your routes file. Rails apps typically use [flash notices and alerts](http://api.rubyonrails.org/classes/ActionDispatch/Flash.html) combined with `render` instead of new error pages. Having tons of non-RESTful endpoints can quickly become hard to maintain and easy to break.